theano/gpuarray/opt.py

```diff
@@ -235,10 +250,10 @@ def local_opt(node):
                     elif isinstance(new_op, (tuple, list)):
                         return [safe_to_cpu(o) for o in new_op]
                     else:  # suppose it is a variable on the GPU
-                        return [host_from_gpu(new_op)]
+                        return [new_op.transfer('cpu')]
             return False
         local_opt.__name__ = maker.__name__
-        return local_optimizer(OP)(local_opt)
+        return register_topo('fast_compile')(local_optimizer(OP)(local_opt))
```

Comment:
I agree this is aimed at introducing a new feature, the backward pass. Please let me know if the plan of having one is still on.